x/crosschain/keeper/keeper_cross_chain_tx.go

```diff
@@ -136,12 +135,30 @@ func (k Keeper) CctxAllPending(c context.Context, req *types.QueryAllCctxPending
 	if !found {
 		return nil, status.Error(codes.Internal, "pending nonces not found")
 	}
+	sends := make([]*types.CrossChainTx, 0)
 
-	if p.NonceLow >= p.NonceHigh {
-		return &types.QueryAllCctxPendingResponse{CrossChainTx: []*types.CrossChainTx{}}, nil
+	// now query the previous nonces up to 100 prior to find any pending cctx that we might have missed
+	// need this logic because a confirmation of higher nonce will automatically update the p.NonceLow
+	// therefore might mask some lower nonce cctx that is still pending.
+	startNonce := p.NonceLow - 100
+	if startNonce < 0 {
+		startNonce = 0
 	}
-	//sends := k.GetAllCctxByStatuses(ctx, []types.CctxStatus{types.CctxStatus_PendingOutbound, types.CctxStatus_PendingRevert})
-	sends := make([]*types.CrossChainTx, 0)
+	for i := startNonce; i < p.NonceLow; i++ {
+		res, found := k.GetNonceToCctx(ctx, tss.TssPubkey, int64(req.ChainId), i)
+		if !found {
+			return nil, status.Error(codes.Internal, fmt.Sprintf("nonceToCctx not found: nonce %d, chainid %d", i, req.ChainId))
+		}
```

Comment:
Can this condition ever happen? 
If it does happen, how to we get out of this? A cctx with an older nonce would need to be created.

Reply:
it should not happen unless serious mess up happened

Reply:
We can just log it then instead of returning , the return will cause clients to be stuck in an infinite loop